### PR TITLE
chore(state): unify data model concerning identities

### PR DESCRIPTION
**Pull request by sdwoodbury**
### What this PR does 📖

- Identities need to be updated with the user's username, status, profile picture, banner, and whether or not they are online. Identities were previously stored in both the friends list and with every chat, and the two lists didn't necessarily overlap. Obviously this is bad. 
- A `HashMap<DID, Identity>` was added to state. everywhere else in state, `Identity` was replaced with `DID`. 
- A few fields in state were made private because altering them may (either now or in the future) require updating `state.identities`. 
- Lots of utility functions were added to state in response to these fields becoming private. 
- If you read this far, good job! 
- There are also miscellaneous changes due to `cargo fmt` and `cargo clippy`

### Which issue(s) this PR fixes 🔨

- Resolve #

### CODE REVIEW

**Review by phillsatellite — APPROVED** [review]

File: common/src/state/action.rs [hunk 1]
```diff
@@ -103,8 +103,8 @@ pub enum Action {
     CancelRequest(Identity),
 
     /// Handle a new incoming friend request
-    #[display(fmt = "IncomingRequest")]
-    IncomingRequest(Identity),
+    // #[display(fmt = "IncomingRequest")]
+    // IncomingRequest(Identity),
```
Comment (InfamousVague) [inline common/src/state/action.rs:107]: why comment this out? Should we just remove it if it's dead code?

File: common/src/state/chats.rs [hunk 2]
```diff
@@ -41,10 +39,10 @@ pub struct Chat {
 // warning: Chats implements Serialize
 #[derive(Clone, Debug, Default, Deserialize)]
 pub struct Chats {
-    #[serde(default)]
+    #[serde(skip)]
     pub initialized: bool,
     // All active chats from warp.
-    #[serde(default)]
+    #[serde(skip)]
```
Comment (InfamousVague) [inline common/src/state/chats.rs:45]: we would want to cache these in state so we don't have to call warp every time I would think, doesn't this break that flow?

**Review by InfamousVague — APPROVED** [review]
Some comments, looks good otherwise

File: common/src/state/chats.rs (hunk 2 above)
Comment (sdwoodbury) [reply common/src/state/chats.rs:45]: warp doesn't have to be called every time, only once on startup...

File: common/src/state/action.rs (hunk 1 above)
Comment (sdwoodbury) [reply common/src/state/action.rs:107]: i didn't know if something was planned for later, but I'm fine with deleting it.

Comment (InfamousVague) [reply common/src/state/action.rs:107]: Let's nuke it 👍

**Pull request merged**